cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -159,7 +159,7 @@ pub trait Environment: Send {
     fn upgrade(
         &mut self,
         flox: &Flox,
-        groups_or_iids: &[String],
+        groups_or_iids: &[&str],
```

Comment:
Suggestion: This seems to require a lot of changes to interfaces and in the end you're converting these `&str` back to `String` anyway.